masschallenge/impact-api
# AC-5058b: Separate "became entrant" events and deeper industry info in GET

Work done:
- "became entrant" events no longer list the set of Programs that a Startup is interested in, but instead there is a separate event for each Program in a Cycle that the Startup is interested in.
- GET calls for user list, user detail, organization list and organization detail provide full Industry details for the "primary_industry" and "additional_industries".  Note: This makes POST and PATCH a bit asymmetrical since they still only take the "primary_industry_id".

To test:
- `make load-remote-db DB_FILE_NAME=initial_schema.sql.gz`
- `make grant-permissions PERMISSION_USER=demoadmin@masschallenge.org PERMISSION_CLASSES=v1_clients`
- Create client application in the new system.
- Best to use Postman.  Remember to update the client_id with the new one from the above.
- Look at the output of the GET options for OrganizationHistory (e.g., http://localhost:8000/api/v1/organization/10542/history/) and note that on the branch there are multiple events for the appropriate cycles, but on development these events are combined through the "programs" key.  In the example, look for "cycle_id": 6.  This startup has two StartupProgramInterest objects for the same Program which is a data problem not an issue addressed by this ticket.
- Test GET, POST, PATCH as appropriate for /v1/user, /v1/user/<id>, /v1/organization, and /v1/organization/<id> with special attention to the primary industry and additional industries related fields.
- Look at the OPTIONS for all the above.  They should accurate reflect the changes, values returned, and the values expected for these calls.
